src/main/java/datawave/user/AuthorizationsListBase.java

```diff
@@ -0,0 +1,342 @@
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlSeeAlso(DefaultAuthorizationsList.class)
+public abstract class AuthorizationsListBase<T> implements Message<T>, HtmlProvider {
+    protected static final String TITLE = "Effective Authorizations", EMPTY = "";
+    
+    @XmlElementWrapper(name = "allAuths")
+    @XmlElement(name = "auth")
+    protected TreeSet<String> userAuths = new TreeSet<String>();
+    
+    protected String userDn;
+    protected String issuerDn;
+    protected List<String> messages = new ArrayList<>();
+    
+    protected LinkedHashMap<SubjectIssuerDNPair,Set<String>> auths = new LinkedHashMap<SubjectIssuerDNPair,Set<String>>();
+    
+    protected Map<String,Collection<String>> authMapping = new HashMap<>();
+    
+    public abstract String getMainContent();
+    
+    @XmlElement(name = "entityAuths")
+    @XmlJavaTypeAdapter(AuthListAdapter.class)
+    public LinkedHashMap<SubjectIssuerDNPair,Set<String>> getAuths() {
+        
+        LinkedHashMap<SubjectIssuerDNPair,Set<String>> authMap = new LinkedHashMap<SubjectIssuerDNPair,Set<String>>();
+        for (Map.Entry<SubjectIssuerDNPair,Set<String>> e : auths.entrySet()) {
+            authMap.put(new SubjectIssuerDNPair(e.getKey().subjectDN, e.getKey().issuerDN), Collections.unmodifiableSet(e.getValue()));
+        }
+        return authMap;
+    }
+    
+    public void addAuths(String dn, String issuerDN, Collection<String> dnAuths) {
+        auths.put(new SubjectIssuerDNPair(dn, issuerDN), new TreeSet<String>(dnAuths));
+    }
+    
+    public void setMessages(List<String> messages) {
+        this.messages.clear();
+        if (messages != null) {
+            this.messages.addAll(messages);
+        }
+    }
+    
+    public void addMessage(String message) {
+        this.messages.add(message);
+    }
+    
+    public void setUserAuths(String userDn, String issuerDn, Collection<String> userAuths) {
+        this.userDn = userDn;
+        this.issuerDn = issuerDn;
+        this.userAuths.addAll(userAuths);
+    }
+    
+    public TreeSet<String> getUserAuths() {
+        return new TreeSet<String>(userAuths);
+    }
+    
+    public TreeSet<String> getAllAuths() {
+        return new TreeSet<String>(userAuths);
+    }
+    
+    public String getUserDn() {
+        // if the userDn is empty but we have auths, then use the first auth entry as the user
+        // this handles the case where the userDn and the subjectDn are not being serialized separately.
+        if (userDn == null && !auths.isEmpty()) {
+            return auths.keySet().iterator().next().subjectDN;
+        }
+        return userDn;
+    }
+    
+    public String getIssuerDn() {
+        // if the issuerDn is empty but we have auths, then use the first auth entry as the user
+        // this handles the case where the userDn and subjectDn are not being serialized separately.
+        if (issuerDn == null && !auths.isEmpty()) {
+            return auths.keySet().iterator().next().issuerDN;
```

Comment:
Is it worth setting issuerDn here?  Same idea for getUserDn above.  I'm not sure how many times this may get called.